plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/HelperCommands.java

```diff
@@ -25,4 +31,43 @@ public static boolean isValidTextResource(XtextResource resource) {
     List<Issue> issues = validator.validate(resource, CheckMode.ALL, null);
     return issues.isEmpty();
   }
+  
+  /**
+   * get the xtext model
+   * 
+   * @param embeddedEditorViewPart
+   *          - this is the embedded editor xtext object
+   * @return - Model the xtext model embedded in this editor view
+   */
+  public static Model getModel(EmbeddedEditorView embeddedEditorViewPart) {
+    TextualScenarioProvider p = embeddedEditorViewPart.getProvider();
+    XtextResource resource = p.getResource();
+    EList<EObject> content = resource.getContents();
+    Model domainModel = null;
+    if (!content.isEmpty() && content.get(0) instanceof Model) {
+      domainModel = (Model) resource.getContents().get(0);
+    } else {
+      EmbeddedEditorInstanceHelper.updateModel("scenario {}");
+      EList<EObject> content1 = resource.getContents();
+      if (!content1.isEmpty() && content1.get(0) instanceof Model) {
+        domainModel = (Model) resource.getContents().get(0);
+      }
+    }
+    return domainModel;
+  }
+
+  /**
+   * clear the elements and participants in the xtext model
+   * 
+   * @param domain model
```

Comment:
Maybe put domainModel instead of domain model.

Reply:
ok